### PR TITLE
Add os-volume_upload_image action to volumeactions

### PR DESCRIPTION
**Pull request by jtopjian**
For #238

### CODE REVIEW

**Comment by jtopjian** [comment]
@jrperritt ping for review

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 1]
```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
 	return
 }
 
+// UploadImage will upload image base on the values in UploadImageOptsBuilder
+func UploadImage(client *gophercloud.ServiceClient, id string, opts UploadImageOptsBuilder) UploadImageResult {
```
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: ```(r UploadImageResult)```

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 2]
```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
...
+	var res UploadImageResult
+
+	reqBody, err := opts.ToVolumeUploadImageMap()
+	if err != nil {
+		res.Err = err
+		return res
+	}
+
+	_, res.Err = client.Post(uploadURL(client, id), reqBody, nil, &gophercloud.RequestOpts{
```
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: `r.Err`

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 3]
```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
 	return
 }
 
+// UploadImage will upload image base on the values in UploadImageOptsBuilder
+func UploadImage(client *gophercloud.ServiceClient, id string, opts UploadImageOptsBuilder) UploadImageResult {
+	var res UploadImageResult
+
+	reqBody, err := opts.ToVolumeUploadImageMap()
```
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: `b, err := `...

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 4]
```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
...
+	if err != nil {
+		res.Err = err
```
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: `r.Err`

File: openstack/blockstorage/extensions/volumeactions/requests.go [hunk 5]
```diff
@@ -88,6 +112,23 @@ func Detach(client *gophercloud.ServiceClient, id string, opts DetachOptsBuilder
...
+		return res
```
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: `return`

File: openstack/blockstorage/extensions/volumeactions/requests.go (hunk 2 above)
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: s/reqBody/b

File: openstack/blockstorage/extensions/volumeactions/requests.go (hunk 1 above)
Comment (jrperritt) [inline openstack/blockstorage/extensions/volumeactions/requests.go]: Let's group all the parts of this operation together (i.e. this function and  `UploadImageOpts` and its methods)

**Comment by jtopjian** [comment]
@jrperritt sorry about that -- I should have done a closer review at the first pass.

ping for second review.

**Comment by jrperritt** [comment]
It's not a problem; thank you for bringing this over.

**Pull request merged**